lib/Factory/Local.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Diversity\Factory;
+
+use Diversity\Component;
+use Diversity\NotFoundException;
+
+use vierbergenlars\SemVer\version;
+use vierbergenlars\SemVer\expression;
+use vierbergenlars\SemVer\SemVerException;
+
+class Local extends \Diversity\Factory {
+  /// Array of Component instances, keyed by spec (uri/name)
+  protected $components = array();
+
+  private $settings = array(
+    'archive'     => null,
+    'archive_url' => null,
+  );
+
+  public function __construct($settings = array()) {
+    $this->settings = array_merge($this->settings, $settings);
+  }
+
+  public function get($component, $version = null) {
+    // Supporting deprecated format of component:version
+    $spec = $component . ($version ? ':' . $version : '');
+
+    if (array_key_exists($spec, $this->components)) return $this->components[$spec];
+
+    $component_data = $this->getComponentData($spec);
+
+    if (!isset($component_data['spec'])) {
+      /// @todo More verbose error message?  More specific exception type.
+      throw new NotFoundException(
+        'Couldn\'t find component: ' . $spec . " in {$this->settings['archive']}"
+      );
+    }
+
+    if (isset($this->settings['archive_url'])) {
+      $component_data['base_url'] = $this->settings['archive_url'] . $component_data['subpath'];
+    }
+
+    $component = new Component($this, $component_data);
+    $this->components[$spec] = $component;
+
+    return $component;
+  }
+
+  /**
+   * Get the asset content.
+   */
+  public function getAsset(Component $component, $asset) {
+    return file_get_contents($this->joinPaths($component->base_dir, $asset));
+  }
+
+  /**
+   * Combine paths handling slashes.
+   *
+   * http://stackoverflow.com/questions/1091107/how-to-join-filesystem-path-strings-in-php/15575293#15575293
+   */
+  private function joinPaths() {
+    $paths = array();
+
+    foreach (func_get_args() as $arg) {
+      if ($arg !== '') { $paths[] = $arg; }
+    }
+
+    return preg_replace('#/+#','/',join('/', $paths));
```

Comment:
I have heard PHP on windows do some magic to resolve this, but is it really a good idea to assume forward slash as the directory separator? Why not use DIRECTORY_SEPARATOR constant instead?